testplan/web_ui/testing/src/AssertionPane/AssertionTypes/GraphAssertions/graphUtils.js

```diff
@@ -27,24 +27,49 @@ export function returnXType(graph_type){
     }
 }
 
+const COLOUR_PALETTE=['#1c5c9c', '#68caea', '#7448c5', '#633836',
+                      '#485051', '#336a85', '#94b1c5', '#ababab']
 /**
- * Return a colour for the graph component depending on whether the user has
- * specified an option, otherwise return random colour (tinted with blue)
+ * Return the colours for every series, given the series_options. Each series'
+ * colour assigned depending on whether the user has specified an option,
+ * otherwise return from a colour scheme/palette, then random colours
+ * (tinted darker/blue)
  */
-export function returnColour(options){
-    if(options == null){
-        let colour = '';
-        for (let i = 0; i < 4; i++) {
-         colour += (Math.round(Math.random() * 15)).toString(16);
+export function returnColour(series_options, data){
+    let series_colours = {};
+    let colour_options = COLOUR_PALETTE.slice(0);
+
+    for(let series_key in data){
```

Comment:
What exactly is `data` here? If it is an array you should use data.forEach() instead.

Reply:
`data` is an object containing the { series_name: data_array[] }, it is here in order to assign a colour to each of the series, it is needed to know every series the user created, (they don't have to create series_options for a series) so data is the only structure that has every series name.

I should make this more clear and should use `Object.keys(data)` and then iteratre through all the series_name like this with .forEach(), i think this is better and will do this :]

Reply:
OK great. I've seen some areas of the code use JSDoc notation to document parameter/return types for functions, maybe that would be useful to add in some places where the types are not obvious.

https://devhints.io/jsdoc